neddf/scripts/fields_visualizer.py

```diff
@@ -0,0 +1,596 @@
+                [-tx_far, ty_far, -d_far],
+                [tx_far, ty_far, -d_far],
+                [tx_far, -ty_far, -d_far],
+                [-tx_far, -ty_far, -d_far],
+            ]
+            edges: List[List[int]] = [
+                [0, 1],
+                [1, 2],
+                [2, 3],
+                [3, 0],
+                [4, 5],
+                [5, 6],
+                [6, 7],
+                [7, 4],
+                [0, 4],
+                [1, 5],
+                [2, 6],
+                [3, 7],
+            ]
+            colors: ndarray = np.ones((12, 3), np.float32)
+            colors[:, 0] = 0.2
+            colors[:, 1] = 0.8
+            colors[:, 2] = 0.4
+
+            lines: o3d.geometry.LineSet = o3d.geometry.LineSet()
+            lines.points = o3d.utility.Vector3dVector(vtx)
+            lines.lines = o3d.utility.Vector2iVector(edges)
+            lines.colors = o3d.utility.Vector3dVector(colors)
+
+            transform: ndarray = np.eye(4)
+            transform[:3, :3] = Rotation.from_rotvec(camera_param[:3]).as_matrix()
+            transform[:3, 3] = camera_param[3:6]
+            lines.transform(transform)
+
+            self.scene.scene.add_geometry(
+                "camera_{}_visible_range".format(idx), lines, self.thin_line_material
+            )
+
+    def generate_mesh(self) -> None:
+        cube_range: float = 1.1
+        mesh_dir: Path = self.output_dir / "mesh"
+        mesh_dir.mkdir(exist_ok=True)
+        voxel_file_path: Path = mesh_dir / "voxel_{}.npy".format(
+            self.meshing_resolution
+        )
+        if voxel_file_path.exists():
+            voxel: ndarray = np.load(voxel_file_path.as_posix())
+        else:
+            voxel = self.trainer.neural_render.get_network().voxelize(
+                field_name="distance",
+                cube_range=cube_range,
+                cube_resolution=self.meshing_resolution,
+            )
+            np.save(voxel_file_path.as_posix(), voxel)
+
+        vertices, triangles = mcubes.marching_cubes(voxel, self.meshing_threshold)
+        vertices -= self.meshing_resolution / 2.0
+        vertices *= 2.0 * cube_range / self.meshing_resolution
+        vertices_list = np.asarray(vertices)
+        triangles_list = np.asarray(triangles)
+        self.meshed_field = o3d.geometry.TriangleMesh(
+            o3d.utility.Vector3dVector(vertices_list),
+            o3d.utility.Vector3iVector(triangles_list),
+        )
+        self.meshed_field.compute_vertex_normals()
+
+        transform: ndarray = np.zeros((4, 4))
+        transform[0, 2] = -1
+        transform[1, 0] = -1
+        transform[2, 1] = 1
+        transform[3, 3] = 1
+        self.meshed_field.transform(transform)
+
+        # mesh_file_path: Path = mesh_dir / "mesh_{}_threshold{}.dae".format(
+        #     cube_resolution, threshold
+        # )
+        # mcubes.export_mesh(vertices, triangles, mesh_file_path.as_posix, "mcube")
```

Comment:
@ueda0319 I think these lines should be un-commented for save exported mesh file.

Reply:
Thank you! I updated it on PR #51 , please confirm.